Userland/Services/WebContent/WebDriverConnection.cpp

```diff
@@ -1272,8 +1289,90 @@ Messages::WebDriverClient::ElementClickResponse WebDriverConnection::element_cli
     // FIXME: 7. If element’s container is obscured by another element, return error with error code element click intercepted.
 
     // 8. Matching on element:
-    // FIXME: option element
-    // FIXME: Otherwise
+    //    - option element
+    if (is<Web::HTML::HTMLOptionElement>(*element)) {
+        auto& option_element = static_cast<Web::HTML::HTMLOptionElement&>(*element);
+
+        // 1. Let parent node be the element’s container.
+        auto parent_node = element_container;
+
+        // 2. Fire a mouseOver event at parent node.
+        fire_an_event<Web::UIEvents::MouseEvent>("mouseOver", parent_node);
+
+        // 3. Fire a mouseMove event at parent node.
+        fire_an_event<Web::UIEvents::MouseEvent>("mouseMove", parent_node);
+
+        // 4. Fire a mouseDown event at parent node.
+        fire_an_event<Web::UIEvents::MouseEvent>("mouseDown", parent_node);
+
+        // 5. Run the focusing steps on parent node.
+        Web::HTML::run_focusing_steps(parent_node.has_value() ? &*parent_node : nullptr);
+
+        // 6. If element is not disabled:
+        if (!option_element.is_actually_disabled()) {
+            // 1. Fire an input event at parent node.
+            fire_an_event<Web::DOM::Event>("input", parent_node);
+
+            // 2. Let previous selectedness be equal to element selectedness.
+            auto previous_selectedness = option_element.selected();
+
+            // 3. If element’s container has the multiple attribute, toggle the element’s selectedness state
+            //    by setting it to the opposite value of its current selectedness.
+            if (parent_node.has_value() && parent_node->has_attribute("multiple")) {
+                option_element.set_selected(!option_element.selected());
+            }
+            //    Otherwise, set the element’s selectedness state to true.
+            else {
+                option_element.set_selected(true);
+            }
+
+            // 4. If previous selectedness is false, fire a change event at parent node.
+            if (!previous_selectedness) {
+                fire_an_event<Web::DOM::Event>("change", parent_node);
+            }
+        }
+        // 7. Fire a mouseUp event at parent node.
+        fire_an_event<Web::UIEvents::MouseEvent>("mouseUp", parent_node);
+
+        // 8. Fire a click event at parent node.
+        fire_an_event<Web::UIEvents::MouseEvent>("click", parent_node);
+    }
+    //    - Otherwise
+    else {
+        // FIXME: 1. Let input state be the result of get the input state given current session and current top-level browsing context.
```

Comment:
I'd just say "FIXME: Implement steps 1-17", but we can also do this :sweat_smile:

Reply:
Originally I was actually going to implement them but I ran out of steam. :sweat_smile: